### ISSUE TITLE
Reformat Selection does something different than reformat document

### ISSUE THREAD

**Issue by sillyotter** [fsprojects/VisualFSharpPowerTools#1332]
When I highlight the contents of a file and tell it to reformat the selection, things seem to work fine.

When I simply request a reformat of the entire document code that looked like this:

 ```fsharp
// Procedures
type ProcedureInfo = 
    { SpecificCatalog : string
      SpecificSchema : string
      SpecificName : string
      RoutineCatalog : string
      RoutineSchema : string
      RoutineName : string
      RoutineType : string // PROCEDURE or FUNCTION
      Created : DateTime
      LastAltered : DateTime }
```

gets reformatted so that all the comments are moved to the top of the document.

I haven't seen any docs that describe this as being expected, nor any way to adjust how to two formats work.  Nor any other issues about it.  Anyone know whats up?

**Comment by cloudRoutine** [comment]
you should report this in https://github.com/dungpa/fantomas/issues

**Comment by dungpa** [comment]
It's fine to open an issue here.

Format Selection creates the AST based on the selection only. It means the scope will be smaller than the big AST from the whole document. One caveat is that F# ASTs don't preserve comments so reattaching comments to formatted code is based on heuristics. Smaller scopes mean less chance to be buggy with regard to comment attachment.

**Comment by sillyotter** [comment]
Thanks for the information.  I want to clarify that the selection in my case was the entire document.  I would have assumed the scope would have been fairly close to the format document scope, but I didn't dig into the code to see what the difference was. (and hence why I didnt send it off to fantomas).

Thanks again.

**Comment by dungpa** [comment]
I'm closing since it's more like a question. If you give a small example, I could look into the deviation between the two features. There might be some oversights in the two implementations. Thank you.

**Comment by sillyotter** [comment]
OK, I'll look into it more myself, but I don't see how this was a question.  I provided a description of a bug and instructions on how to recreate it that anyone could follow with any code that had comments in it.  When it seemed you weren't clear on what I said, I tried to provide more information.  I'm assuming that anyone interested in this has some code laying around with some comments in it. and could compare the difference between format document and select all format selection.  If it didn't do the same for you, then that would tell us something.

In any event, thanks for your help, if I dig up anything ill let you know.
